src/mono/System.Private.CoreLib/src/System/Threading/ThreadPool.Browser.Mono.cs

```diff
@@ -118,5 +116,36 @@ private static void Callback()
             _callbackQueued = false;
             ThreadPoolWorkQueue.Dispatch();
         }
+
+        private static unsafe void NativeOverlappedCallback(nint overlappedPtr) =>
```

Comment:
CoreCLR throws PNSE for `UnsafeQueueNativeOverlapped` so I kept that behavior for the portable implementation. It looks like the implementation may have been taken from the NativeAOT implementation, where it had a common implementation between Windows (where that implementation is necessary) and on Unix (where it could throw PNSE). I kept the implementations the same as before for now for those cases that don't use the portable thread pool. So on Mono now it would throw PNSE on Unixes. I think they could all throw PNSE on Unixes, not sure if the minor change in behavior would matter.